inception-concept-linking/src/main/java/de/tudarmstadt/ukp/inception/conceptlinking/service/ConceptLinkingService.java

```diff
@@ -468,4 +470,12 @@ private SemanticSignature getSemanticSignature(KnowledgeBase aKB, String aWikida
             .collect(Collectors.toList());
     }
 
+    @EventListener
+    public void onKnowledgeBaseConfigurationChangedEvent(
+        KnowledgeBaseConfigurationChangedEvent aEvent)
+    {
+        candidateCache.clear();
```

Comment:
If the caches are cross-project and the projects have different KBs using same IRIs but different values for them, then the cache would leak across projects. 

I believe the caches must be per-project and when receiving the event, only the cache for the respective project must be clear. In that case, there must also be some time-based cache-expunge strategy since we can have arbitrary numbers of project and if there is no automatic cache clearing, then the caches could easily consume lots of memory and starve the system.

An alternative would be to include the project in the cache key and when clearing the caches on a KB change, drop only those cache entries with a matching project. In this case, all projects share a common cache which is memory-bounded and wouldn't cause the system to starve even if not regularly cleared up.

The second alternative is probable the better one.